interfaces/user_controller_test.go

```diff
@@ -0,0 +1,350 @@
+		Age:              0,
+		Gender:           0,
+		ImageUrl:         "",
+		FavoriteMusicAge: 0,
+		FavoriteArtist:   "",
+		Comment:          "",
+	}
+
+	users := []model.User{user1, user2}
+
+	return &users, nil
+}
+
+func (fur *FakeUserRepository) GetUser(userEmail string) (*model.User, error) {
+	user := model.User{
+		ID:               1,
+		CreatedAt:        time.Date(2020, 6, 1, 9, 0, 0, 0, time.Local),
+		UpdatedAt:        time.Date(2020, 6, 1, 9, 0, 0, 0, time.Local),
+		Password:         "aaaaaa",
+		Name:             "",
+		Email:            "a@test.co.jp",
+		Age:              0,
+		Gender:           0,
+		ImageUrl:         "",
+		FavoriteMusicAge: 0,
+		FavoriteArtist:   "",
+		Comment:          "",
+	}
+
+	return &user, nil
+}
+
+func (fur *FakeUserRepository) FindByID(userID int) (*model.User, error) {
+	user := model.User{
+		ID:               1,
+		CreatedAt:        time.Date(2020, 6, 1, 9, 0, 0, 0, time.Local),
+		UpdatedAt:        time.Date(2020, 6, 1, 9, 0, 0, 0, time.Local),
+		Password:         "aaaaaa",
+		Name:             "",
+		Email:            "a@test.co.jp",
+		Age:              0,
+		Gender:           0,
+		ImageUrl:         "",
+		FavoriteMusicAge: 0,
+		FavoriteArtist:   "",
+		Comment:          "",
+	}
+
+	return &user, nil
+}
+
+func (fur *FakeUserRepository) Update(userID int, p model.User) error {
+	return nil
+}
+
+//idで指定したユーザーの情報を返すハンドラのテスト
+func TestGetUserHandler(t *testing.T) {
+	req := httptest.NewRequest("GET", "/api/user/1", nil)
+
+	//headerをセット
+	req.Header.Set("Content-Type", "application/json")
+
+	//リクエストユーザー作成
+	user := model.User{Email: "a@test.co.jp", Password: "aaaaaa"}
+
+	//トークン作成
+	token, err := createToken(user)
+	if err != nil {
+		log.Println("err:", err)
+	}
+
+	jointToken := "Bearer" + " " + token
+
+	req.Header.Set("Authorization", jointToken)
+
+	// テスト用のレスポンス作成
+	res := httptest.NewRecorder()
+
+	fakeUserController := &UserController{
+		UserInteractor: usecases.UserInteractor{
+			UserRepository: &FakeUserRepository{},
+		},
+	}
+
+	r := mux.NewRouter()
+	r.Handle("/api/user/{id}", http.HandlerFunc(fakeUserController.GetUserHandler)).Methods("GET")
+	r.ServeHTTP(res, req)
+
+	// レスポンスのステータスコードのテスト
+	if res.Code != http.StatusOK {
+		t.Errorf("invalid code: %d", res.Code)
+	}
+
+	//レスポンスボディをDecode
+	var p model.User
+	dec := json.NewDecoder(res.Body)
+	if err := dec.Decode(&p); err != nil {
+		log.Println(err)
+	}
+
+	//期待値(アサート用の構造体)
+	expected := model.User{
+		ID:               1,
+		CreatedAt:        time.Date(2020, 6, 1, 9, 0, 0, 0, time.Local),
+		UpdatedAt:        time.Date(2020, 6, 1, 9, 0, 0, 0, time.Local),
+		Name:             "",
+		Email:            "a@test.co.jp",
+		Age:              0,
+		Gender:           0,
+		ImageUrl:         "",
+		FavoriteMusicAge: 0,
+		FavoriteArtist:   "",
+		Comment:          "",
+	}
+
+	// レスポンスのボディが期待通りか確認
+	if diff := cmp.Diff(p, expected); diff != "" {
+		t.Errorf("handler returned unexpected body: got %v want %v",
+			p, expected)
```

Comment:
diff を print するようにするといいと思います

```suggestion
		t.Errorf("handler returned mismatch (-want +got):\n%s", diff))
```